src/components/marker-chart/Canvas.js

```diff
@@ -114,9 +118,27 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
           ? undefined
           : markerIndexToTimingRow.get(hoveredItem);
 
-      if (newRow !== undefined) {
+      if (newRow !== undefined && newRow !== rightClickedRow) {
```

Comment:
I think we can remove this second condition and have some simpler code.

Initial code was:
```js
if (there's a new hovered item) {
  // Redraw the row for this new hovered item -- so that it's highlighted
}
if (there was an old hovered item, on a different row) {
  // redraw the row for the old item -- so that it's back to normal
}
```

So I believe the new code could be something like this:
```js
if (there's a new hovered item) {
  // 1. clear this row
  // (no need to draw separators and labels, it's been done in clear row)
  // 2. highlight row (always)
  // 3. draw markers for this row
}
if (there was an old hovered item, on a different row) {
  // 1. clear this row
  // (no need to draw separators and labels, it's been done in clear row)
  // 2. if (right clicked marker row is this row) highlight row
  // 3. draw markers for this row
}
```

Reply:
Maybe you can add these comments as well, to make the code easier to follow. I had to think a lot to find what we were doing here :-)

Reply:
Yes sure, this seems a more elegant way to go about things.